highfidelity/hifi
# WL 21369 - Implement parts of the JS "console" object

https://worklist.net/21369

**Description:**
Implemented the following methods for the High Fidelity script engine:

```
console.info()
console.log()
console.debug()
console.warn()
console.error()
console.exception()

console.trace()
console.clear() // Only for the Scripting Console

console.asserts()

console.group()
console.groupCollapsed()
console.groupEnd()

console.time()
console.timeEnd()
```

**Function descriptions:**
https://developer.mozilla.org/en-US/docs/Web/API/Console

We have created test file called "consoleObjectTest.js" to test above functions.

**How to test ?**
1) Open debug window from interface menu Developer -> script logs.
2) press ctrl + j in desktop mode.
3) Type "consoleObjectTest.js" in filter textbox.
4) Select "consoleObjectTest.js" and Press Load button.
5) Click on refresh icon of "consoleObjectTest.js" in currently running scripts window.
6) Observe output of each function of above list with proper tags in debug window.
7) Observe similar output in Logs/log file.
